rust/osm2lanes/src/transform/tags_to_lanes/lane_count.rs

```diff
@@ -0,0 +1,375 @@
+                    LANES + "backward",
+                    CENTRE_TURN_LANE,
+                ])));
+            }
+
+            if let (Some(l), Some(f)) = (tagged_lanes, tagged_forward) {
+                // TODO What is the right warning for straight up conflicts in tag values?
+                if l != f {
+                    warnings.push(TagsToLanesMsg::ambiguous(
+                        &format!("{l}!={f}"),
+                        tags.subset(&[ONEWAY, LANES, LANES + "forward"]),
+                    ));
+                }
+            }
+
+            let assumed_forward = 1; // TODO depends on highway tag
+            Self {
+                lanes: tagged_lanes.map_or_else(
+                    || tagged_forward.map_or(Infer::Default(assumed_forward), Infer::Direct),
+                    Infer::Direct,
+                ),
+                forward: tagged_forward.map_or_else(
+                    || {
+                        tagged_lanes.map_or(Infer::Default(assumed_forward), |l| {
+                            Infer::Direct(l.checked_sub(pre_backward).unwrap())
+                        })
+                    },
+                    Infer::Direct,
+                ),
+                backward: tagged_backward.map_or(Infer::Default(pre_backward), Infer::Direct),
+                both_ways,
+            }
+        } else {
+            // Not oneway
+            match (tagged_lanes, tagged_forward, tagged_backward) {
+                (Some(l), Some(f), Some(b)) => {
+                    if l != f
+                        .checked_add(b)
+                        .unwrap()
+                        .checked_add(both_way_lanes)
+                        .unwrap()
+                    {
+                        warnings.push(TagsToLanesMsg::ambiguous_tags(tags.subset(&[
+                            "lanes",
+                            "lanes:forward",
+                            "lanes:both_ways",
+                            "lanes:backward",
+                        ])));
+                    }
+                    Self {
+                        lanes: Infer::Direct(l),
+                        forward: Infer::Direct(f),
+                        backward: Infer::Direct(b),
+                        both_ways,
+                    }
+                },
+                (None, Some(f), Some(b)) => Self {
+                    lanes: Infer::Calculated(
+                        f.checked_add(b)
+                            .unwrap()
+                            .checked_add(both_way_lanes)
+                            .unwrap(),
+                    ),
+                    forward: Infer::Direct(f),
+                    backward: Infer::Direct(b),
+                    both_ways,
+                },
+                (Some(l), Some(f), None) => Self {
+                    lanes: Infer::Direct(l),
+                    forward: Infer::Direct(f),
+                    backward: Infer::Calculated(
+                        l.checked_sub(f)
```

Comment:
(`checked_sub` here does make sense, though I don't think we want to panic if something says `lanes=2, lanes:backward=5`)

Reply:
I will add a TODO + expect for now